File: LeetCode/Solution_217.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        s = set()
        for num in nums:
            if num in s:
                return True
            s.add(num)
        return False
    
# Time Complexity: O(n) - We traverse the list of numbers once
# Space Complexity: O(n) - In the worst case, we may store all numbers in

File: LeetCode/Solution_217.java
// 217. Contains Duplicate

import java.util.*;

class Solution_217 {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> seen = new HashSet<>();
        for(int num: nums) {
            if (seen.contains(num)) {
                return true;
            }
            seen.add(num);
        }
        return false;
    }
}

// Time Complexity: O(n)
// Space Complexity: O(n)